Sources/Account/Account.swift

```diff
@@ -72,14 +88,17 @@ final class Account {
         return (txOuts: txOuts, blockCount: knowableBlockCount)
     }
 
-    var unspentTxOuts: [KnownTxOut] {
-        unspentTxOutsAndBlockCount.txOuts
+    func unspentTxOuts(tokenId: TokenId) -> [KnownTxOut] {
+        unspentTxOutsAndBlockCount(tokenId: tokenId).txOuts
     }
 
-    var unspentTxOutsAndBlockCount: (txOuts: [KnownTxOut], blockCount: UInt64) {
+    func unspentTxOutsAndBlockCount(
+        tokenId: TokenId
+    ) -> (txOuts: [KnownTxOut], blockCount: UInt64) {
         let knowableBlockCount = self.knowableBlockCount
         let txOuts = allTxOutTrackers
             .filter { $0.receivedAndUnspent(asOfBlockCount: knowableBlockCount) }
+            .filter { $0.knownTxOut.tokenId == tokenId }
```

Comment:
This can be done in one filter pass with && instead of having to do two passes, but I like the way the code looks... 

Whether or not it matters kind of depends on how large the initial array (and filtered array) might be, and how often this is called.

Reply:
I prefer the readability, if these become bottlenecks we know how to improve but I don't think they are right now.

Reply:
verified not a bottleneck